src/packages/core/src/utils/create-or-update-entities.ts

```diff
@@ -125,32 +127,69 @@ export const createOrUpdateEntities = async <G extends WithId, D extends BaseDat
 			const isRelatedEntity = relatedEntity && relatedEntity.prototype instanceof GraphQLEntity;
 
 			if (isRelatedEntity) {
-				// We have a related entity so lets check if we need to do an update or create
-				const result = isLinking(childNode) // if all entities are in the format {id: ""} then we are just linking the entity
-					? childNode
-					: await callChildMutation(
-							getMutationName(relatedEntity.name, childNode),
-							childNode,
-							info,
-							context
-					  );
-				node = {
-					...node,
-					[key]: result,
-				};
+				if (isLinking(childNode)) {
+					// If it's a linking entity or an array of linking entities, nothing to do here
+				} else if (Array.isArray(childNode)) {
+					// If we have an array, we may need to create the parent first as children need reference to the parent
+
+					// As are updating the parent from the child, we can remove this key
+					delete node[key as keyof Partial<G>];
+
+					// Check if we already have the parent ID
+					let parentId = node.id ?? parent?.id;
+					if (!parentId && !parent) {
+						// If there's no ID, create the parent first
+						const parentDataEntity = await meta.provider.createOne(node);
+						parent = fromBackendEntity(parentDataEntity, gqlEntityType);
+						parentId = parent?.id;
+					}
+					if (!parentId) {
+						throw new Error(`Implementation Error: No parent id found for ${relatedEntity.name}`);
+					}
+
+					// Add parent ID to children and perform the mutation
+					const childMeta = getMeta(relatedEntity.name);
+					const parentField = childMeta.fields.find((field) => field?.getType() === gqlEntityType);
```

Comment:
Maybe for later, but can we rename getType() to getGqlEntityType?

Reply:
yeah we should review that when we review the whole metadata piece 👍